src/Tokenizer/Tokens.php

```diff
@@ -516,19 +516,43 @@ public function getNextNonWhitespace(int $index, ?string $whitespaces = null): ?
     }
 
     /**
-     * Get index for closest next token of given kind.
+     * Get index for closest next token among given kinds.
      *
      * This method is shorthand for getTokenOfKindSibling method.
      *
-     * @param int $index token index
-     * @param list<array{int}|string|Token> $tokens        possible tokens
-     * @param bool $caseSensitive perform a case sensitive comparison
+     * @param int                                          $index         token index
+     * @param list<array{0: int, 1?: string}|string|Token> $tokens        possible tokens
+     * @param bool                                         $caseSensitive perform a case sensitive comparison
      */
     public function getNextTokenOfKind(int $index, array $tokens = [], bool $caseSensitive = true): ?int
     {
         return $this->getTokenOfKindSibling($index, 1, $tokens, $caseSensitive);
     }
 
+    /**
+     * Get index for closest next token of given kind.
+     *
+     * @param int                                        $index         token index
+     * @param array{0: int, 1?: string}|int|string|Token $tokens        possible tokens
+     * @param bool                                       $caseSensitive perform a case sensitive comparison
+     *
+     * @internal
```

Comment:
For now those new methods are only here to ease writing type-safe code. However I think they would be a nice addition (replacement?) to current API but I'm not sure about their signature, hence the `@internal` flag. WDYT?

Reply:
I'm not sure if I understand the signature difference of input params. what's the change and benefit?

Reply:
The only difference in parameters is that `$tokens` is not a list of token kinds/prototypes but a single one. I think in most cases we actually need to look for a single kind of token but the current method needs it to be wrapped in an array. It also accepts a single token id. So instead of passing `[[T_FOO]]` we can now pass `T_FOO`. By the way, `$tokens` should be renamed to `$token`.

I also thought about dropping the `$caseSensitive` parameter in favor of a distinct method, but I kept it for now.

Reply:
to be honest I'm not sure if we should add more and more methods into `Tokens` class. maybe we should aim for dedicated traversable/search classes instead?